### ISSUE TITLE
Explicit package version requirement check does not account for %_isa (and maybe transitive exact deps)

### ISSUE THREAD

**Issue by hroncok** [rpminspect/rpminspect#613]
I've just seen the following warnings in https://src.fedoraproject.org/rpms/pypy3.8/pull-request/2

:warning: Subpackage pypy3.9 on x86_64 carries 'Requires: libpypy3.9-c.so.0.1()(64bit)' which comes from subpackage pypy3.9-libs but does not carry an explicit package version requirement. Please add 'Requires: pypy3.9-libs = %{version}-%{release}' to the spec file to avoid the need to test interoperability between various combinations of old and new subpackages.

:warning: Subpackage pypy3.9-devel on x86_64 carries 'Requires: libpypy3.9-c.so.0.1()(64bit)' which comes from subpackage pypy3.9-libs but does not carry an explicit package version requirement. Please add 'Requires: pypy3.9-libs = %{version}-%{release}' to the spec file to avoid the need to test interoperability between various combinations of old and new subpackages.

Yet the packages have:

```
$ rpm -q --requires -p pypy3.9-7.3.8~rc1-1.fc37.x86_64.rpm 
...
libpypy3.9-c.so.0.1()(64bit)
pypy3.9-libs(x86-64) = 7.3.8~rc1-1.fc37
```

Clearly `pypy3.9-libs(x86-64) = 7.3.8~rc1-1.fc37` is *stronger* than plain `pypy3.9-libs = 7.3.8~rc1-1.fc37`. Hence the first report should not warn me. In fact, the explicit requirement must be arched, so the remedy might even suggest that.

And:

```
$ rpm -q --requires -p pypy3.9-devel-7.3.8~rc1-1.fc37.x86_64.rpm 
...
libpypy3.9-c.so.0.1()(64bit)
pypy3.9(x86-64) = 7.3.8~rc1-1.fc37
...
```

Here we see
 - the same problem
 - also pypy3.9-devel requires exact pypy3.9 requires exact pypy3.9-libs, so the dependency is exact yet transitive

I cannot tell if transitive dep would otherwise be satisfactory for the check, due to the %_isa problem, but it should work.

**Comment by dcantrell** [comment]
This appears to be fixed with the patch I committed yesterday to address the %{_isa} handling.  I just tried it out with the latest build of rpminspect and the messages you reported with pypy are not showing up.
